recipes/android-ndk/all/conanfile.py

```diff
@@ -0,0 +1,212 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class AndroidNDKInstallerConan(ConanFile):
+    name = "android-ndk"
+    description = "The Android NDK is a toolset that lets you implement parts of your app in " \
+                  "native code, using languages such as C and C++"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://developer.android.com/ndk/"
+    topics = ("NDK", "android", "toolchain", "compiler")
+    license = "Apache-2.0"
+    short_paths = True
+    no_copy_source = True
+    exports_sources = ["cmake-wrapper.cmd", "cmake-wrapper"]
+
+    settings = {"os": ["Windows", "Linux", "Macos"],
```

Comment:
Shouldn't this also include `compiler`, and force it to be `clang`?

Reply:
Clang (from the ndk) is chosen for you anyway, there is no way around to use a different compiler

Reply:
My profile can potentially be configured wrong and set to gcc.
This recipe should also set the clang version.

Reply:
about the compiler version, you get what is in the ndk, and since this recipe is supposed to work with multiple ndk versions ....., hardcoding something in is not productive, I need a better way.

I might be able to validate that the used profile setting is not wrong, in respect to compiler name and version, 
And throw and exception if there is a wrong compiler name / version in the used host profile.
The name is easy, but with the version I currently do not know how to do. Ideas welcome !

Reply:
Is the compiler from the Conan profile used at all or only the NDK included Clang?

If only the included Clang is used the compiler in the profile is irrelevant and should be delete in `package_id()`

Reply:
the compiler on the build machine is irrelevant, there does not even need to be one 
the compiler on the host machine is set via this package,  , and I think  @madebr made a good point that checking the host profile settings against the reality in this package can be valuable.

Since the compiler is not part of settings, I do not need to delete it, or ? 

What is of course missing is that this package can , when be used , add info to the package id, but I guess I need to file a issue for that

Reply:
I would do the following:
- `settings = "os", "arch", "compiler"`
- in `configure`: assert that `os=Android`, `arch=armXXX` and `compiler=clang`, `compiler.libcxx=libc++` and `compiler.version=X`

I think the mingw of conan-community is a good example:
https://github.com/conan-community/conan-mingw-installer/blob/develop/conanfile.py

Reply:
how shall this work? configure runs on the build machine, there are not android infos available
android comes only in play when it consumes the package as a build_requirement, and there is it still in the build context, just the settings_target are Android relevant, in the package_info call

Reply:
Indeed, incomplete/bad suggestion by me.

So the package_id should include: `os`, `arch`, `compiler`, `os_build`, `arch_build` where:
- `os_target=Android`,
- `arch_target=armXX`,
- `compiler=clang`, `compiler.version=X`, `compiler.libcxx=libc++`,
- `os in ["Windows", "Macos", "Linux"]` and
- `arch="x86_64"`

Am I correct to say that the current `settings` is incomplete?

Let's ask our cross-build-in-house-specialist @jgsogo

Reply:
`os_target` and `arch_target` do not exist anymore, at least not here.
this package is designed to work only with the 2 profile approach `--profile::build ... --profile:host` , since this is already working wonderful and this is also the future.

all checks have to happen in  `def package_info(self)` , there I can check that the consumer (`self.settings_target`) has sane settings.
I do this already, check `package_info`
```
        # And if we are not building for Android, why bother at all
        if not self.settings_target.os == "Android":
            return
```

This is basically just a service to the user, to validate the used profile, but I think this is a good place to add this info.
And providing a message is nicer than doing nothing.
However, as mentioned, I have problems to verify clang version, without maintaining a lookup table (which is currently not a problem since there is only 1 version, but I do not like hard coded magic numbers)

Reply:
When do compiler versions change?
Do they change between major versions? Between minor releases?

Reply:
this is irrelevant, since you do not build anything from that package, and it does not need a compiler to be built
this package is providing something,  that between the thousands of things is also a compiler, ...... 

so you can build the package on a system that does not have a compiler and upload it to your remote, works wonderful  

the compiler version is relevant for packages who get compiled using that package as build requirement.
I hope that this point is now clear, and explains why I mark that discussion as resolved

Reply:
This package comes with a compiler. So it is relevant.

Reply:
in best case as an info what it provides. but it requires nothing in the settings.  hope this is now clear

Reply:
I still think `compiler` should be added, just for checking whether:
`self.settings.compiler == "clang"` and optionally `tools.Version(self.settings.compiler.version) == "X.Y"`.

Reply:
one more explanation trial: This recipe is a build requirement, it is build in the build context, with the profile of the build context, and this will be the same profile that,  for example cmake uses, that uses openssl....
You CAN NOT USE `settings.compiler` for that package in that context, and for sure not set it to clang, because it is very likely (100%) something different, but you do not care since there is nothing to build.
I hope this explains it now, @madebr

Reply:
Ok, I guess you mean the following:
this recipe is meant to be used in the host profile as a build requirement.
So settings.compiler in this context means compiler to build for host=Linux/Windows/Macos.
What it needs to check instead is `target_compiler``.

Am I correct in saying this?